counsel.el

```diff
@@ -5073,6 +5073,50 @@ in the current window."
             :unwind #'counsel--switch-buffer-unwind
             :update-fn 'counsel--switch-buffer-update-fn))
 
+
+;;** `counsel-compile'
+(defvar counsel-compile-history nil
+  "History for `counsel-compile'.")
+
+(defvar counsel-compile-local-builds "make -k"
+  "Additional compile invocations to feed into `counsel-compile'.
+
+This can either be a list of compile invocations strings or
+functions that will provide such a list.  You should customise
+this if you want to provide specific non-standard build types to
+`counsel-compile'.  The default helpers are setup to handle common
+build environments.")
+
+(defun counsel--get-compile-candidates ()
+  "Return the list of compile commands as directed by
+`counsel-compile-local-builds'."
+  (let ((cands))
+    (if (stringp counsel-compile-local-builds)
+        (setq cands (list counsel-compile-local-builds))
+      (mapc
+       (lambda (c)
+         (let ((more-cands
+                (cond
+                 ((functionp c) (funcall c))
+                 ((stringp c) (list c))
+                 ((listp c) c))))
+           (when more-cands
+             (setq cands (append cands more-cands)))))
```

Comment:
I don't like that `append` will be repeatedly copying previous entries potentially several times.  Find a way to construct the complete list in one shot (only one copy per element).

Reply:
What would you recommend as a good form for this?

Reply:
I guess whenever it's your private list constructed here, aka `(list c)`, `nconc` it with `cands`, while when it's a list returned from elsewhere (e.g. function), `nconc` a copy of it.